cfgov/unprocessed/js/routes/on-demand/simple-chart/simple-chart.js

```diff
@@ -206,6 +209,105 @@ function makeFormatter( yAxisLabel ) {
   };
 }
 
+
+/**
+ * Extracts all dates from an object/csv formatted for tilemap display
+ * @param {object} data The data object
+ * @returns {array} Extracted dates
+ * */
+function getTilemapDates( data ) {
+  return Object.keys( data[0] )
+    .filter( k => !isNaN( new Date( k ) ) )
+    .sort( ( a, b ) => new Date( b ) - new Date( a ) );
+}
+
+/**
+ * Builds the tilemap filter DOM
+ * @param {object} chartNode The node where the chart loives
+ * @param {object} chart The chart object
+ * @param {object} data The data object
+ * @param {object} transform Whether data has been transformed
+ */
+function makeTilemapSelect( chartNode, chart, data, transform ) {
+
+  let d;
+  if ( transform ) d = data.transformed;
+  else d = data.raw;
+
+  const options = getTilemapDates( d );
+  const selectNode = makeFilterDOM( options, chartNode, { key: 'tilemap' },
+    'Select a date'
+  );
+
+  attachTilemapFilter( selectNode, chart, data );
+}
+
+
+/**
+ * Wires up the tilemap filter
+ * @param {object} node The created select node
+ * @param {object} chart The chart object
+ * @param {object} data The data object
+ */
+function attachTilemapFilter( node, chart, data ) {
+  node.addEventListener( 'change', evt => {
+    const formatted = formatSeries( data );
+    const updated = getMapConfig( formatted, evt.target.value );
+    chart.update( updated );
+  } );
+}
+
+
+/**
+ * Adds generates a config object to be added to the chart config
+ * @param {array} series The formatted series data
+ * @param {string} date The date to use
+ * @returns {array} series data with a geographic component added
+ * */
+function getMapConfig( series, date ) {
+  let min = Infinity;
+  let max = -Infinity;
+  const data = series[0].data;
+  if ( !date ) date = getTilemapDates( data )[0];
+  const added = data.map( v => {
+    const val = Math.round( Number( v[date] ) * 100 ) / 100;
+    if ( val <= min ) min = val;
+    if ( val >= max ) max = val;
+    return {
+      ...usLayout[v.state_ab],
+      state: v.state_ab,
+      value: val
+    };
+  } );
+  min = Math.floor( min );
+  max = Math.ceil( max );
+  const step = Math.round( ( max - min ) / 4 );
+  return {
+    colorAxis: {
+      dataClasses: [ {
+        from: min,
+        to: min + step,
+        color: '#e2efd8',
+        name: `${ min } - ${ min + step }`
+      }, {
+        from: min + step,
+        to: min + ( 2 * step ),
```

Comment:
Consider assigning these calculations to named variables to make this code clearer. Something like `group_2_max`, `group_3_max`, etc.

Reply:
Fixed in faa3997e0